### ISSUE TITLE
[BUG] prepareFEP is broken with recent versions of BSS

### ISSUE THREAD

**Issue by jmichel80** [OpenBioSim/biosimspace#88]
**Describe the bug**
Using 

Sire version: 2022.3.0 and BSS  version '2022.2.1+351.gf2538d20'

The command below run on the attached input completes succesfully. 
[inputs.zip](https://github.com/OpenBioSim/biosimspace/files/11593903/inputs.zip)

`python prepareFEP.py --input1 thrombin_truncated_1a_sort.prm7 thrombin_truncated_1a_sort.rst7 --input2 thrombin_truncated_1b_sort.prm7 thrombin_truncated_1b_sort.rst7 --output 1a~1b`

When using
Sire version: 2023.2.2  and BSS version '2023.3.0.dev+25.g945779ae'

The above command gives

```
  File "/home/matthew/jm/prepareFEP/prepareFEP.py", line 17, in <module>
    from Sire.Mol import AtomIdx
ModuleNotFoundError: No module named 'Sire'
```
Replacing the above statement with

`from sire.mol import AtomIdx `

and running again gives

```
Cannot import sire using the mixed API as either the old or new APIs have already been activated.
(...)
AttributeError: type object 'MoleculeParser' has no attribute 'supportedFormats'
```

Swapping the order of the import statements from

```
from sire.mol import AtomIdx
import BioSimSpace as BSS
```

to

```
import BioSimSpace as BSS
from sire.mol import AtomIdx
```
Resolves the issue. 

Not sure I understand why. If the above fix is safe then I suggest updating the version of prepareFEP at

`https://github.com/OpenBioSim/biosimspace/tree/devel/nodes/playground` 

to be compatible with devel .

**Comment by chryswoods** [comment]
This is because of the way I updated from the old to new sire APIs, while  still keeping a backwards-compatibility option for scripts that need to use the old API.

Sire can be loaded in one of three ways

## 1. Using the new API

This should be preferred for all new scripts, except for those that also need to include BioSimSpace (see 3 below)

```
import sire as sr
sr.use_new_api()
```

Or the new API will be automatically loaded if you use any new API functions, e.g.

```
import sire as sr
mols = sr.load(sr.expand(sr.tutorial_url, "ala.top", "ala.crd"))
```

will automatically load the new API when you call `sr.load`.

The old modules are available in `sire.legacy`, e.g. `sire.legacy.Mol` is the old `Sire.Mol`.

Note however that all of the function names will be fully pythonic, i.e. `mol.num_atoms()` rather than `mol.nAtoms()` etc.

## 2. Using the old API

This is for backwards compatibility for old sire scripts so that they don't need porting.

```
try:
    import sire as sr
    sr.use_old_api()
except ImportError:
    # We really are using the old version of sire
    pass

# can now import Sire as before
import Sire
from Sire.Mol import AtomIdx
# etc.
```

This works by plugging into the python module import system and creating a fake `Sire` module with fake paths that redirect to the legacy modules in `sire.legacy`. We had to use this route, rather than allow `Sire` and `sire` to be loaded at the same time, because the MacOS file system is not case-sensitive - we can't have a directory called `Sire` and `sire` that are different...

Note that the old, non-pythonic function names are still used, and the new pythonic ones aren't available (you need to use `mol.nAtoms()` not `mol.num_atoms()` etc.

## 3. Using the mixed API

```
import sire as sr
sr.use_mixed_api()
```

The worst option, but what we need for BioSimSpace while it is ported, is to have both the new and old APIs loaded simultaneously. This provides both the `Sire` and `sire` modules at the same time, and objects have both the pythonic and non-pythonic functions (so you can use both `mol.nAtoms()` and `mol.num_atoms()`).

We use this so that we can write new BioSimSpace functionality using the new API, while keeping the old functionality in the old API. Eventually we will port everything over to the new API and will deprecate the old API (over a long time period).

There isn't a rush to do this, as the old and new APIs use the same code underneath - it is all just renaming from a C++ naming style to a pythonic naming style, and adding new functionality that takes advantage of the greater adoption of python.

In your case,

```
from Sire.Mol import AtomIdx
import BioSimSpace as BSS
```

failed as there is no `Sire` module, because `sire` hasn't been imported and the old (or mixed) API hasn't been selected.

Then

```
from sire.mol import AtomIdx
import BioSimSpace as BSS
```

failed because importing `AtomIdx` from `sire.mol` used the new API, so sire automatically actives the new API. However, BioSimSpace needs the mixed API, and this can't be activated after the new or old APIs have been activated (you have to choose one at the start of the script).

```
import BioSimSpace as BSS
from sire.mol import AtomIdx
```

worked because BioSimSpace has internally imported sire and activated the mixed API. Further use of sire is now in the mixed API.

TL;DR - always make sure that you import BioSimSpace in a script before importing sire. This will ensure that sire is loaded with the correct API mode.

**Comment by lohedges** [comment]
I'll think of a place to add this to the documentation. Perhaps the developer's guide?
